integrations/terraform/msk-privatelink/README.md

```diff
@@ -0,0 +1,122 @@
+
+The end result is a Materialize cluster that is connected to an Amazon MSK cluster via an AWS PrivateLink connection.
+
+## Prerequisites
+
+- [Terraform](https://www.terraform.io/downloads.html) 0.13 or later.
+- [Materialize Cloud](https://cloud.materialize.com/) account.
+- [AWS](https://aws.amazon.com/) account.
+- [MSK cluster](https://docs.aws.amazon.com/msk/latest/developerguide/create-cluster.html).
+- Follow the steps in the [README.md](../README.md) file to set up your AWS configuration and Materialize details.
+
+## Overview of the Terraform Configuration
+
+The configuration is divided into several sections:
+
+1.  **Define the Materialize provider**: This section specifies the required Materialize provider version and its source.
+2.  **Include the AWS provider**: This section configures the AWS provider with the specified region.
+3.  **Include the Materialize provider**: This section configures the Materialize provider with connection information, such as host, username, password, port, and database.
+4.  **Use the Materialize MSK module**: This section sets up the MSK cluster by providing necessary information such as the cluster name, port, VPC ID, and AWS region.
+5.  **Create a PrivateLink connection in Materialize**: This section creates a PrivateLink connection in Materialize by providing the necessary information such as name, schema name, service name, and availability zones.
+6.  **Add the Materialize allowed principal to the AWS VPC Endpoint Service**: This section allows the Materialize principal to access the AWS VPC Endpoint Service by providing the VPC Endpoint Service ID and Materialize principal ARN.
+7.  **Approve the VPC Endpoint Service connection**: This section instructs you to manually approve the VPC Endpoint Service connection in your AWS account.
+
+## Step-by-step Instructions
+
+### Step 1: Define the Materialize provider
+
+This block specifies the required version of the Materialize provider and its source:
+
+```hcl
+terraform {
+  required_providers {
+    materialize = {
+      source = "MaterializeInc/materialize"
+      version = "0.0.3"
+    }
+  }
+}
+```
+
+### Step 2: Include the AWS provider
+
+Configure the AWS provider with the specified region:
+
+```hcl
+provider "aws" {
+    region = "us-east-1"
+}
+```
+
+### Step 3: Include the Materialize provider
+
+Configure the Materialize provider with the necessary connection information:
+
+```hcl
+provider "materialize" {
+  host     = local.materialize_host
+  username = local.materialize_username
+  password = local.materialize_password
+  port     = 6875
+  database = "materialize"
+}
+```
+
+
+### Step 4: Use the Materialize MSK module
+
+Set up the MSK cluster by providing the necessary information:
+
+```hcl
+module "msk" {
+  source              = "MaterializeInc/msk-privatelink/aws"
+  version             = "0.1.2"
+  mz_msk_cluster_name = "example-msk-cluster"
```

Comment:
Should the MSK variables also be supplied via the `local` for consistency with how they are set for the Materialize?

Reply:
Neat! Good catch!